edb/server/compiler/compiler.py

```diff
@@ -536,7 +536,17 @@ def parse_search_path(search_path_str: str) -> list[str]:
         sql_units = []
         for stmt in stmts:
             # GOTCHA: a setting is frontend-only regardless of its mutability
-            fe_only = stmt.name in fe_settings_mutable
+            if isinstance(
+                stmt,
+                (
+                    pgast.VariableSetStmt,
+                    pgast.VariableResetStmt,
+                    pgast.VariableShowStmt,
+                ),
+            ):
+                fe_only = stmt.name in fe_settings_mutable
```

Comment:
Ah the `stmt` may not have a `name` 🤦‍♂️ thanks for the fix! Tho perhaps we can push this variable into the `if` branches as it's only needed there